### ISSUE TITLE
TV series become unwatchable when they have too many episodes

### ISSUE THREAD

**Issue by maybeanerd** [jellyfin/jellyfin-web#5779]
**Describe The Bug**
When starting an episode in a series with many episodes (e.g. one piece, 1110 episodes) the playback is stuck in the loading (before opening player) state forever.
This behavior does not occur when I remove 800 of the 1110 episodes and try to play the _exact same_ episode, so it seemed to be related to the amount of total episodes. This also happens on all episodes I tried, not only specific ones.

When digging deeper, I found that on server side, no errors are being thrown, but on client side, there is a request to get all episodes (https://jellyfin.cluster.diluz.io/Shows/cfe.../Episodes?IsVirtualUnaired=false&IsMissing=false&UserId=93d..&Fields=Chapters,Trickplay) which is huge. For one piece, it ends up being 2.4MB and takes quite some time to finish.

The client seems to be unhappy about the duration of the request, as it logs `Request timed out`
![image](https://github.com/user-attachments/assets/10dfe1a2-5eba-4792-9a77-96a027f6e424)

although in the network tab, the request finishes with 200 (but maybe is stopped early by the client? the response is truncated)
![image](https://github.com/user-attachments/assets/907f0bfe-524a-4485-9f3d-11b6332ca662)

**Steps To Reproduce**

1. Add a series with approx. 1000 episodes
2. Let library scan finish (important! it doesn't happen if the library scan hasn't added all of the episodes yet)
3. Try to play one of the episodes
4. It will load forever

**Expected Behavior**
The episode should start playing (even if it takes long, which I would not expect, but tolerate)

**Logs**
[jellyfin.log](https://github.com/user-attachments/files/16201945/jellyfin.log)

**Screenshots**
This is what it will look like forever
![image](https://github.com/user-attachments/assets/0e2c37d8-38b0-4295-aae0-e4650b480a90)

**System (please complete the following information):**
 - Platform: Web (Firefox), WebOs (LG TV app), Android App , even third party apps (although If this is a client side issue, I will need to open bug reports there as well.)
 - Browser: Firefox, but also native apps as listed above
 - Jellyfin Version: 10.9.7 on web&server

**Additional Context**
none

**Comment by maybeanerd** [comment]
I'm going to try to adjust some things in the client experimentally to see if that would actually fix the entire issue - will need to first figure out where even to look
If in the meantime a maintainer sees this, feel free to drop some extra info and context on this behavior please

**Comment by maybeanerd** [comment]
first attempts seem to not break any existing behavior, but stop the probelmatic call from fetching _all_ episodes: https://github.com/jellyfin/jellyfin-web/compare/master...maybeanerd:jellyfin-web:fix/playback-of-series-with-large-set-of-episodes?expand=1#diff-b338dd22dfaa8db18edede53fe56aabb5d911c5e611cc33ad3c914410d021cbf

I have yet to test this with the 1000+ episode setup, though.
Will do so soon™️ to ensure that this actually fixes the cause of the loading issue

**Comment by maybeanerd** [comment]
Tested the solution, works! It fixes the issue for 1110 episodes.
I'll open it as a PR, where I would hope someone who knows more about the system itself can check if this would break other existing behavior. I did try around everything I could think of and nothing seemed broken

**Comment by maybeanerd** [comment]
Could actually be that https://github.com/Namo2/InPlayerEpisodePreview/pull/33 solved this issue?
I'll check if I can still reproduce it

**Comment by alimbada** [comment]
How are your episodes structured? Are they all in one folder on the server?

**Comment by alimbada** [comment]
Interesting. I have the same setup but don't experience this issue. I wonder if it's due to how your library is set up?

This is how mine's set up:

![Image](https://github.com/user-attachments/assets/b0d9135a-8384-478f-ab7e-5520fede32af)

![Image](https://github.com/user-attachments/assets/80851979-8c74-4405-8491-b614ac8b208f)

![Image](https://github.com/user-attachments/assets/b9ea21bf-e11b-4ebd-8ee4-deaeff9fc216)

**Comment by maybeanerd** [comment]
Can you trace the the request your client does when loading an episode? As mentioned in the original issue text, the request seems to not fail, instead it is a 200 that is probably cancelled due to a client side timeout.

Could be that your hardware is so much faster than mine that the timout doesn't hit?


In any case, I'll add back the content (removed tons of intermittent seasons for it to work) and see if I have the same issue with version 10.10.1 as well

**Comment by maybeanerd** [comment]
I was still able to reproduce the issue, but in the PR that is linked to this issue, I think I now finally have provided a usable solution.

**Comment by alimbada** [comment]
> Can you trace the the request your client does when loading an episode? As mentioned in the original issue text, the request seems to not fail, instead it is a 200 that is probably cancelled due to a client side timeout.

Yep, I get a 1.83MB response which is truncated (I have ~100 fewer episodes than you do). It took 1.27s to fetch it.

> Could be that your hardware is so much faster than mine that the timout doesn't hit?

Client I used for checking the above is Firefox + Windows 10 on a Ryzen 7 3700X machine with 16GB RAM but I also use the Android TV app on Chromecast with Google TV, the LG webOS app on an LG C2 and the iOS app on an iPhone 11.

Server is a NanoPi R6C running Armbian.

I just had a look at your PR and it seems there are some limitations with your new implementation though I'm not too familiar with the Jellyfin API or frontend but if there's any issues raised by the maintainers I can look into making the necessary API changes to expedite a better solution if needed.
